sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -805,6 +805,7 @@ public void tearDown() {
     private final Instant currentTimestamp;
     private final Duration allowedLateness;
     private final WindowedValue<?> currentElementOrTimer;
+    private Instant currentOutputTimestamp = null;
```

Comment:
```suggestion
    @Nullable private Instant currentOutputTimestamp;
```

Reply:
Nit: non-final fields are null by default. The annotation probably helps to indicate the default null value.